src/sync.cpp

```diff
@@ -672,7 +630,15 @@ namespace librealsense
                 //LOG_IF_ENABLE( "...     next expected of the missing stream didn't updated yet", env );
                 return false;
             }
-            LOG_IF_ENABLE( "...     exceeded {10*gap}" << threshold, env );
+            LOG_IF_ENABLE( "...     exceeded threshold of {10*gap}" << threshold << "; deactivating matcher!", env );
+
+            auto const q_it = _frames_queue.find( missing );
+            if( q_it != _frames_queue.end() )
+            {
+                if( q_it->second.empty() )
+                    _frames_queue.erase( q_it );
+            }
+            missing->set_active( false );
```

Comment:
timestamp = 66 next_expected =67
in that case you will set_active( false ) while actually not needed.

Reply:
As discussed -- won't get here with these numbers